File: AtCoder/ACP/BootCamp4b/Hard100_2/python/C_Inserting_x.py
S = input()
left = 0
right = len(S) - 1
res = 0
while left < right:
    if S[left] == S[right]:
        left += 1
        right -= 1
    else:
        if S[left] == 'x':
            left += 1
            res += 1
        elif S[right] == 'x':
            right -= 1
            res += 1
        else:
            res = -1
            break
print(res)

File: AtCoder/ACP/BootCamp4b/Hard100_2/cpp/C_Inserting_x.cpp
#include <bits/stdc++.h>
using namespace std;
typedef long long ll;

string S;

void solve() {
  int left = 0;
  int right = S.size() - 1;
  ll res = 0;
  while (left < right) {
    if (S[left] == S[right]) {
      ++left;
      --right;
    } else {
      if (S[left] == 'x') {
        ++res;
        ++left;
      } else if (S[right] == 'x') {
        ++res;
        --right;
      } else {
        res = -1;
        break;
      }
    }
  }
  printf("%lld\n", res);
}

int main() {
  cin >> S;
  solve();
}